MM-31749 - Adjust documentation of Mattermost Operator (#4281)
mattermost/docs: source/install/install-kubernetes-mattermost.rst

Deploying a Mattermost Installation
===================================

.. warning::

>   If you used the Mattermost Operator in version prior to v1.12.x or still using ``ClusterInstallation`` Custom Resource 

Reviewer: ```suggestion
  If you used the Mattermost Operator in version prior to v1.12.x or are still using ``ClusterInstallation`` Custom Resource 
```

  check out `this guide <https://github.com/mattermost/mattermost-operator/blob/master/docs/migration.md>`__ to see how to migrate to new ``Mattermost`` resource.
  
  The ``ClusterInstallation`` is deprecated and will be removed in version v2.0.

> This guide describes deploying a complete Mattermost installation in Kubernetes. 

Reviewer: ```suggestion
This guide describes deploying a complete Mattermost installation in Kubernetes.
```

> Manifest files contain the configurations needed for the operator to properly setup the Mattermost installation. 

Reviewer: ```suggestion
Manifest files contain the configurations needed for the Operator to properly set up the Mattermost installation.
```

Create the manifest files locally in a text editor, copy and paste the contents, and save the file. Recommended file names are provided, but your naming conventions may differ.

> Manifesets are applied with ``kubectl``. Before running the commands make sure you are connected to your Kubernetes cluster.

Reviewer 2: ```suggestion
Manifests are applied with ``kubectl``. Before running the commands make sure you are connected to your Kubernetes cluster.
```

    name: mattermost-license
  type: Opaque
  stringData:
    license: [LICENSE_FILE_CONTENTS]

Save the file as ``mattermost-license-secret.yaml``.

**2. Create an installation manifest file**

> The Mattermost installation manifest contains fields which must be adjusted for your configuration and environment requirements. 

Reviewer: ```suggestion
The Mattermost installation manifest contains fields which must be adjusted for your configuration and environment requirements.
```

> Some of the most commonly used fields are:

Reviewer: ```suggestion
Some of the most commonly-used fields are:
```

    "spec.version", "The Mattermost version.", "No"
    "spec.licenseSecret", "The name of the Kubernetes secret containing your license (e.g. mattermost-license). Required for Enterprise deployments.", "No"
    "spec.mattermostEnv", "List of custom environment variables for the Mattermost instance.", "No"
    
More fields are documented `in the example <https://github.com/mattermost/mattermost-operator/blob/master/docs/examples/mattermost_full.yaml>`__.
If you have previous experience with Kubernetes Custom Resources you can also check the `Custom Resource Definition <https://github.com/mattermost/mattermost-operator/blob/master/config/crd/bases/installation.mattermost.com_mattermosts.yaml>`__.

Open a text editor and create a Mattermost installation manifest:

.. code-block:: yaml

  apiVersion: installation.mattermost.com/v1beta1
  kind: Mattermost
  metadata:
    name: mm-example-full                         # Chose the desired name
  spec:
    size: 5000users                               # Adjust to your requirements
    ingressName: example.mattermost-example.com   # Adjust to your domain
    ingressAnnotations:
      kubernetes.io/ingress.class: nginx

>     version: 5.28.0

Reviewer 2: ```suggestion
    version: 5.31.0
```

Latest version

    licenseSecret: ""                             # If you have created secret in step 1, put its name here
    
Save the file as ``mattermost-installation.yaml``.

.. warning::

>   To configure the Mattermost Installation with an external database and filestore refer to `Configuring external database and filestore (Recommended)`_ before proceeding.

Reviewer 2: I wonder if it would be better to just show how to configure external db and filestore in the step above. Then in this warning/note say you can use the mysql/minio operators but they're not recommended for production. What do you think?

Author: I merged the guide of configuring external db and filestore with the main one and added note, that those can be skipped.
Let me know what do you think.

>   To configure the Mattermost Installation with an external database and filestore refer to `Configuring external database and filestore (Recommended)`_ before proceeding.

Reviewer: ```suggestion
  To configure the Mattermost installation with an external database and filestore refer to `Configuring external database and filestore (Recommended)`_ before proceeding.
```

    
More fields are documented `in the example <https://github.com/mattermost/mattermost-operator/blob/master/docs/examples/mattermost_full.yaml>`__.
If you have previous experience with Kubernetes Custom Resources you can also check the `Custom Resource Definition <https://github.com/mattermost/mattermost-operator/blob/master/config/crd/bases/installation.mattermost.com_mattermosts.yaml>`__.

Open a text editor and create a Mattermost installation manifest:

.. code-block:: yaml

  apiVersion: installation.mattermost.com/v1beta1
  kind: Mattermost
  metadata:
    name: mm-example-full                         # Chose the desired name
  spec:
    size: 5000users                               # Adjust to your requirements
    ingressName: example.mattermost-example.com   # Adjust to your domain
    ingressAnnotations:
      kubernetes.io/ingress.class: nginx
    version: 5.31.0
    licenseSecret: ""                             # If you have created secret in step 1, put its name here
    
Save the file as ``mattermost-installation.yaml``.

.. note::
    Steps 3 to 5 cover configuring Mattermost with external database and filestore which is recommended installation configuration. 

>     When using MySQL and MinIO operators this steps can be skipped. 

Reviewer 2: ```suggestion
    When using MySQL and MinIO operators these steps can be skipped. 
```
